templates/sql_database/vault.tf

```diff
@@ -0,0 +1,11 @@
+resource "azurerm_key_vault_secret" "db_username" {
+  key_vault_id = var.vault_id
+  name         = "${var.env}-db-username"
+  value        = var.administrator_login
+}
+
+//NOTE: This will need to be manually created in your key vault before running terraform.
```

Comment:
Is something stopping us from creating this resource in Terraform?

Reply:
Primarily time and additional testing. 😅 I didn't get the chance to go in and verify how this would be done, or how rotation would end up working. Plus, doing it in this manner gives the end user of these templates the ability to conform whatever password is needed to their own internal requirements, rather than us doing it for them. (That's the story I will stick to, anyway!)

This is absolutely something we can address and make better with an enhancement PR.